src/delivery.cpp

```diff
@@ -0,0 +1,180 @@
+std::string receiverLocation{};
+
+geometry_msgs:: Vector3 setpointArray;
+
+ros::Publisher availabilityPub{};
+ros::Publisher progressStatusPub{};
+ros::Publisher setpointPub{};
+
+void assignSetpoint(double placeArray[3])
+{
+    setpointArray.x = placeArray[0];
+    setpointArray.y = placeArray[1];
+    setpointArray.z = placeArray[2];
+}
+
+void setAvailabilityStatus(std::string status)
+{
+    std_msgs::String availabilityStatus;
+    std::stringstream ssAvailabilityStatus;
+    ssAvailabilityStatus << status;
+    availabilityStatus.data = ssAvailabilityStatus.str();
+
+    availabilityPub.publish(availabilityStatus);
+}
+
+void setProgressStatus(std::string status)
+{
+    std_msgs::String progressStatus;
+    std::stringstream ssProgressStatus;
+    ssProgressStatus << status;
+    progressStatus.data = ssProgressStatus.str();
+
+    progressStatusPub.publish(progressStatus);
+}
+
+bool findSetpoint(std::string location)
+{
+    if (location.compare("Location A") == 0)
+    {
+        assignSetpoint(placeA);
+    }
+    else if (location.compare("Location B") == 0)
+    {
+        assignSetpoint(placeB);
+    }
+    else if (location.compare("Location C") == 0)
+    {
+        assignSetpoint(placeC);
+    }
+    else if (location.compare("Location D") == 0)
+    {
+        assignSetpoint(placeD);
+    }
+    else
+    {
+        assignSetpoint(placeNil);
+        return 0;
+    }
+
+    return 1;
+}
+
+void senderLocationCallback(const std_msgs::String::ConstPtr &name)
+{
+    senderLocation = name->data.c_str();
+}
+
+void receiverLocationCallback(const std_msgs::String::ConstPtr &name)
+{
+    receiverLocation = name->data.c_str();
+}
+
+void reachedSetpointCallback(const std_msgs::Bool::ConstPtr &value)
+{   
+    reachedSetpointBool = value->data;
+}
+
+int main(int argc, char **argv)
+{
+    ros::init(argc,argv,"delivery");
+    ros::NodeHandle n;
+
+    // Publishers
+    availabilityPub = n.advertise<std_msgs::String>("availability",1000);
+    progressStatusPub = n.advertise<std_msgs::String>("progress",1000);
+    setpointPub = n.advertise<geometry_msgs::Vector3>("setpoint",1000);
+
+    // Subscribers
+    ros::Subscriber senderLocationSub = n.subscribe("senderLocation",1000,senderLocationCallback);
+    ros::Subscriber receiverLocationSub = n.subscribe("receiverLocation",1000,receiverLocationCallback);
+    ros::Subscriber reachedSetpointSub = n.subscribe("reachedSetpointBool",10,reachedSetpointCallback);
+
+    ros::Rate loopRate(10); 
+
+    while (ros::ok())
+    {
+        // Sender loop
+        int senderloopCount{0};
+        while (ros::ok())
+        {
+            bool foundSetpointBool{findSetpoint(senderLocation)};            
+            setpointPub.publish(setpointArray);
+
+            if (foundSetpointBool)
+            {
+                setAvailabilityStatus("no");
+                setProgressStatus("in progress");
+            }
+
+            if (senderloopCount++ == 0) {sleep(2);}
```

Comment:
either use braces in the next line or dump all the commands in a single line. Not both! Not like this!